kubernetes/kubernetes_meter.py

```diff
@@ -44,13 +44,17 @@ def _parse_apel_str(self, apel_record: str):
         self.apel_dict = {}
         lines = apel_record.split('\n')
         for line in lines:
-            kv_pair = [v.strip() for v in line.split(':')]
-            if len(kv_pair) == 2:
-                self.apel_dict[kv_pair[0]] = kv_pair[1]
+            # Handle case where apel values contain colons (eg. 'SubmitHost: http://localhost:8000')
+            kv_pair_match = re.match(r'([A-Za-z0-9_]+)\s*:\s*([^\s].*[^\s])\s*',line)
+            if kv_pair_match:
+                self.apel_dict[kv_pair_match[1]] = kv_pair_match[2]
 
     def getint(self, key):
         return int(float(self.apel_dict.get(key, 0)))
 
+    def getint_roundup(self, key):
+        return int(float(self.apel_dict.get(key, 0)) + 0.5)
+
```

Comment:
Does this actually work? 

```
blin@blin-work:~$ python
Python 3.12.3 (main, Apr 17 2024, 00:00:00) [GCC 13.2.1 20240316 (Red Hat 13.2.1-7)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> int(float(3.9))
3
```

Are you looking for `math.ceiling()` instead?

Reply:
This rounds >= X.5 to X +1, <= X.5 to X, eg:
```
>>> int(float("3.9") + .5)
4
>>> int(float("3.4") + .5)
3
```

. math.ceiling() is probably more appropriate though

Reply:
Ah duh, that makes sense. `math.ceiling` doesn't do any sort of rounding so maybe what you had was right?

I'm not actually sure I understand what the ultimate goal is here.

Reply:
CPU requests are configured as 0.9 and 7.9 for 1 and 8 core jobs respectively in Armen's cluster to allow other small pods to run on the same node, which were getting truncated to 0 or 7 by this code. Either math.ceil or the +.5 trick will round them to 1 and 8 as desired in this case. I think math.ceil is slightly preferable since it will round a processor count of <0.5 to 1 instead of 0, but that is probably a rare corner case.

Reply:
Huh and why do we need to turn this into an int?

Reply:
That is a good question. I looked at several sample gratia records while implementing this and noticed that all fields were integers, but never confirmed that that was actually a requirement